.github/workflows/bazel-lockfile.yml

```diff
@@ -0,0 +1,220 @@
+          BUILDEVENT_DATASET: "github-ci-dfinity"
+        with:
+          BAZEL_CI_CONFIG: "--config=ci --config macos_ci"
+          BAZEL_COMMAND: build
+          BAZEL_EXTRA_ARGS: '--test_tag_filters=test_macos --lockfile_mode=update --nobuild'
+          BAZEL_STARTUP_ARGS: "--output_base /var/tmp/bazel-output/${CI_RUN_ID}"
+          BAZEL_TARGETS: "//rs/... //publish/binaries/..."
+          BUILDEVENT_APIKEY: ${{ secrets.HONEYCOMB_API_TOKEN }}
+      - uses: actions/upload-artifact@v4
+        with:
+          name: bazel-module-lock-apple-intel
+          path: ./MODULE.bazel.lock
+
+  # Finally download the lockfile that went through every platform and check it
+  # for changes. See steps for actual logic depending on GHA event.
+  bazel-bzlmod-lockfile-update:
+    name: Check lockfile changes
+    needs: bazel-bzlmod-lockfile-macos-intel # the last platform
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref || github.ref }}
+          # set the token in case we need to push the updated lockfile
+          token: ${{ secrets.IDX_PUSH_TO_PR }}
+      - uses: actions/download-artifact@v4
+        with:
+          name: bazel-module-lock-apple-intel
+      - name: check and push
+        run: |
+          echo git status
+          git status
+          echo checking state
+
+          if git diff --quiet; then
+            # if diff is clean, then the lockfile did not need updating
+            echo "MODULE.bazel.lock is up to date"
+            exit 0
+          fi
+
+          # on merge groups & on master, fail the job
+
+          if [ "${{ github.event_name }}" == "merge_group" ]; then
+            echo "denying merge because of out-of-date MODULE.bazel.lock"
+            exit 1
+          fi
+
+          if [ "${{ github.event_name }}" == "push" ] && \
+              [ "${{ github.ref_name }}" == "master" ]; then
+            echo "MODULE.bazel.lock is out of date on master branch"
+            exit 1
+          fi
+
+          # on dev branches and PRs, update the lockfile
+
+          if [ "${{ github.event_name }}" == "push" ] && \
+              [[ "${{ github.ref_name }}" =~ ^dev-gh-* ]]; then
+
+            echo "updating MODULE.bazel.lock"
+            git config user.email "infra+github-automation@dfinity.org"
+            git config user.name "IDX GitHub Automation"
+            git add MODULE.bazel.lock
+            git commit -m "Update MODULE.bazel.lock"
+            git push
+
+            exit 1
+          fi
+
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+
+            echo "updating MODULE.bazel.lock"
+            git config user.email "infra+github-automation@dfinity.org"
+            git config user.name "IDX GitHub Automation"
+            git add MODULE.bazel.lock
+            git commit -m "Update MODULE.bazel.lock"
+            git push
+
+            exit 1
```

Comment:
Why `exit 1`?

Reply:
because the commit should be marked as red if changes are needed (the lockfile should always be up to date)

Reply:
Not sure I understand the logic. CI job successfully completes updating the lock file when you bump some version, so CI job and entire workflow should / could be green. In most of the cases, if not in all, the workflow is interrupted and commit doesn't end up green.